x-pack/plugins/monitoring/public/lib/get_safe_for_external_link.ts

```diff
@@ -24,15 +24,16 @@ export function getSafeForExternalLink(
 
   let newGlobalState = globalStateExecResult[1];
   Object.keys(globalState).forEach((globalStateKey) => {
+    let value = globalState[globalStateKey];
+    if (globalStateKey === 'cluster_uuid') {
```

Comment:
Where are you seeing the double quotes happening? I'm trying to reproduce it locally but can't seem to find the right UI flow

Reply:
Here is an example: https://demo.elastic.co/app/monitoring#/home
- Notice how clicking on the first cluster has no quotes `...(cluster_uuid:C7AppFTyQfWdUL3ldutLjQ,`
- And clicking on the second cluster has single quotes `...(cluster_uuid:'0f0rwRiWTjaJPTrORm4PoA',`

Since, we don't take "checking first" into account we run the risk of adding unwanted quotes

Reply:
Oh, I'm a little confused sorry. Why are you testing this PR on the demo site? Are you connecting your local Kibana instance to that cluster or something?

Locally, with two clusters setup, I get from clicking on one cluster:
https://localhost:5601/app/monitoring#/overview?_g=(ccs:Monitoring,cluster_uuid:'1sO4J1fWQEq8AU628IoBGw',refreshInterval:(pause:!t,value:10000),time:(from:now-1h,to:now))

Then, I see https://localhost:5601/app/monitoring#/overview?_g=(refreshInterval:(pause:!t,value:10000),time:(from:now-1h,to:now),cluster_uuid:'TpRkL2e-QVaJisp084RwzQ') when clicking the other.

I can see in the code how we could be double quoting it, but I can't seem to find the use case of it actually happening (so I can verify the fix works)

Reply:
Ah, I see it now "globalState" here is not actually the globalState we're use to, more of a way to override parameters (since we only use it one place).

In which case why not just pass the following cluster_uuid already as a quoted string here:
https://github.com/elastic/kibana/blob/3f97872055ef4afa810a5bd633c4fb6e2d21244a/x-pack%2Fplugins%2Fmonitoring%2Fpublic%2Fcomponents%2Fcluster%2Flisting%2Flisting.js#L82 so it'd look something like:
```javascript
getSafeForExternalLink(`#/overview`, { cluster_uuid: `'${cluster.cluster_uuid}'` })
```
This way we don't have to check which key is which and how to cast it.

Might be more of nit/preference on my end, so I'm fine if you decide to keep it. Let me know and I'll approve it

Reply:
Fair enough. I worry that requiring the consumer to provide the `'` will lead to us forgetting to do that. I'm not sure if there is a better way right now though.